src/main/java/command/Parser.java

```diff
@@ -0,0 +1,240 @@
+package command;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import storage.Storage;
+
+import task.Deadline;
+import task.Event;
+import task.ToDo;
+import task.Task;
+import task.TaskList;
+
+/**
+ * Handles the parsing and execution of user commands in the ChatterBox chatbot.
+ * It processes the input commands and delegates the execution to the respective
+ * methods, modifying the tasklist and updating the storage as necessary.
+ */
+public class Parser {
+    //Formatter for displaying dates in the desired format
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mm a");
+
+    /**
+     * Parses and executes user command based on the input string.
+     *
+     * @param input The command input from the user.
+     * @param taskList The list of tasks managed by the chatbot.
+     * @param storage The storage handler responsible for saving tasks.
+     * @param ui The user interface handler for displaying messages.
+     */
+    public void parseExecute(String input, TaskList taskList, Storage storage, Ui ui) {
+        if(input.startsWith("mark")) {
```

Comment:
Take note on single space within the statements. Here, there should be a space between `if` and the condition.